pkg/build/controller/image_change_controller_test.go

```diff
@@ -0,0 +1,121 @@
+	controller := mockImageChangeController(buildcfg, "testImageRepo", "registry.com/namespace/imagename", map[string]string{"test": "newImageId123"})
+	controller.HandleImageRepo()
+	buildCreator := controller.BuildCreator.(*mockBuildCreator)
+	if buildCreator.buildcfg == nil {
+		t.Errorf("New build not created when new image was created")
+	}
+	if buildCreator.imageSubstitutions["registry.com/namespace/imagename"] != "registry.com/namespace/imagename:newImageId123" {
+		t.Errorf("Image substitutions not properly setup for new build: %s |", buildCreator.imageSubstitutions["registry.com/namespace/imagename"])
+	}
+
+	// valid configuration using default latest tag, new build should be triggered.
+	buildcfg = mockBuildConfig("registry.com/namespace/imagename", "registry.com/namespace/imagename", "testImageRepo", "")
+	controller = mockImageChangeController(buildcfg, "testImageRepo", "registry.com/namespace/imagename", map[string]string{"latest": "newImageId123"})
+	controller.HandleImageRepo()
+	buildCreator = controller.BuildCreator.(*mockBuildCreator)
+	if buildCreator.buildcfg == nil {
+		t.Errorf("New build not created when new image was created")
+	}
+	if buildCreator.imageSubstitutions["registry.com/namespace/imagename"] != "registry.com/namespace/imagename:newImageId123" {
+		t.Errorf("Image substitutions not properly setup for new build using default latest tag: %s |", buildCreator.imageSubstitutions["registry.com/namespace/imagename"])
+	}
+
+	// this buildconfig references a non-existent imagerepo, so an update to the real imagerepo should not
+	// trigger a build here.
+	buildcfg = mockBuildConfig("registry.com/namespace/imagename", "registry.com/namespace/imagename", "testImageRepo2", "test")
+	controller = mockImageChangeController(buildcfg, "testImageRepo", "registry.com/namespace/imagename", map[string]string{"test": "newImageId123"})
+	controller.HandleImageRepo()
+	buildCreator = controller.BuildCreator.(*mockBuildCreator)
+	if buildCreator.buildcfg != nil {
+		t.Errorf("New build created when a different repository was updated")
+	}
+
+	// this buildconfig references a different tag than the one that will be updated, this will (for now) result
+	// in a build being triggered, but there should be no image name substitution since the imagerepo does not contain
+	// a valid imageid for the "test2" tag, so we should use the existing image name in the buildconfig.
+	buildcfg = mockBuildConfig("registry.com/namespace/imagename", "registry.com/namespace/imagename", "testImageRepo", "test2")
+	controller = mockImageChangeController(buildcfg, "testImageRepo", "registry.com/namespace/imagename", map[string]string{"test": "newImageId123"})
+	controller.HandleImageRepo()
+	buildCreator = controller.BuildCreator.(*mockBuildCreator)
+	if buildCreator.buildcfg == nil {
+		t.Errorf("New build not created when a different repository tag was updated")
```

Comment:
I agree we should not run a build, but currently we will run a build because we can't distinguish which tag was updated.  that's what my comment is saying...  also my TODO statement in image_change_controller.